### ISSUE TITLE
k-means : WARN storage.MemoryStore: Not enough space to cache rdd

### ISSUE THREAD

**Issue by Jetsonpaul** [intel-hadoop/HiBench#152]
15/11/25 15:59:44 WARN spark.SparkConf: The configuration key 'spark.kryoserializer.buffer.mb' has been deprecated as of Spark 1.4 and may be removed in the future. Please use spark.kryoserializer.buffer instead. The default value for spark.kryoserializer.buffer.mb was previously specified as '0.064'. Fractional values are no longer accepted. To specify the equivalent now, one may use '64k'.
15/11/25 15:59:44 WARN spark.SparkConf: The configuration key 'spark.kryoserializer.buffer.mb' has been deprecated as of Spark 1.4 and and may be removed in the future. Please use the new key 'spark.kryoserializer.buffer' instead.
15/11/25 15:59:44 INFO spark.SparkContext: Running Spark version 1.5.1
15/11/25 15:59:44 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
15/11/25 15:59:45 WARN spark.SparkConf: The configuration key 'spark.kryoserializer.buffer.mb' has been deprecated as of Spark 1.4 and may be removed in the future. Please use spark.kryoserializer.buffer instead. The default value for spark.kryoserializer.buffer.mb was previously specified as '0.064'. Fractional values are no longer accepted. To specify the equivalent now, one may use '64k'.
15/11/25 15:59:45 WARN spark.SparkConf: The configuration key 'spark.kryoserializer.buffer.mb' has been deprecated as of Spark 1.4 and and may be removed in the future. Please use the new key 'spark.kryoserializer.buffer' instead.
15/11/25 15:59:45 INFO spark.SecurityManager: Changing view acls to: root
15/11/25 15:59:45 INFO spark.SecurityManager: Changing modify acls to: root
15/11/25 15:59:45 INFO spark.SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: Set(root); users with modify permissions: Set(root)
15/11/25 15:59:45 INFO slf4j.Slf4jLogger: Slf4jLogger started
15/11/25 15:59:45 INFO Remoting: Starting remoting
15/11/25 15:59:46 INFO Remoting: Remoting started; listening on addresses :[akka.tcp://sparkDriver@10.88.67.113:37244]
15/11/25 15:59:46 INFO util.Utils: Successfully started service 'sparkDriver' on port 37244.
15/11/25 15:59:46 INFO spark.SparkEnv: Registering MapOutputTracker
15/11/25 15:59:46 WARN spark.SparkConf: The configuration key 'spark.kryoserializer.buffer.mb' has been deprecated as of Spark 1.4 and may be removed in the future. Please use spark.kryoserializer.buffer instead. The default value for spark.kryoserializer.buffer.mb was previously specified as '0.064'. Fractional values are no longer accepted. To specify the equivalent now, one may use '64k'.
15/11/25 15:59:46 WARN spark.SparkConf: The configuration key 'spark.kryoserializer.buffer.mb' has been deprecated as of Spark 1.4 and and may be removed in the future. Please use the new key 'spark.kryoserializer.buffer' instead.
15/11/25 15:59:46 WARN spark.SparkConf: The configuration key 'spark.kryoserializer.buffer.mb' has been deprecated as of Spark 1.4 and may be removed in the future. Please use spark.kryoserializer.buffer instead. The default value for spark.kryoserializer.buffer.mb was previously specified as '0.064'. Fractional values are no longer accepted. To specify the equivalent now, one may use '64k'.
15/11/25 15:59:46 WARN spark.SparkConf: The configuration key 'spark.kryoserializer.buffer.mb' has been deprecated as of Spark 1.4 and and may be removed in the future. Please use the new key 'spark.kryoserializer.buffer' instead.
15/11/25 15:59:46 INFO spark.SparkEnv: Registering BlockManagerMaster
15/11/25 15:59:46 INFO storage.DiskBlockManager: Created local directory at /tmp/blockmgr-e22824d0-afcf-4118-af73-31b74f9a2a77
15/11/25 15:59:46 INFO storage.MemoryStore: MemoryStore started with capacity 2.2 GB
15/11/25 15:59:46 WARN spark.SparkConf: The configuration key 'spark.kryoserializer.buffer.mb' has been deprecated as of Spark 1.4 and may be removed in the future. Please use spark.kryoserializer.buffer instead. The default value for spark.kryoserializer.buffer.mb was previously specified as '0.064'. Fractional values are no longer accepted. To specify the equivalent now, one may use '64k'.
15/11/25 15:59:46 WARN spark.SparkConf: The configuration key 'spark.kryoserializer.buffer.mb' has been deprecated as of Spark 1.4 and and may be removed in the future. Please use the new key 'spark.kryoserializer.buffer' instead.
15/11/25 15:59:46 INFO spark.HttpFileServer: HTTP File server directory is /tmp/spark-dbfe2bf6-5ef3-4285-9771-2a6379624fe5/httpd-031ed014-5298-46e3-9369-efc41244ab3c
15/11/25 15:59:46 INFO spark.HttpServer: Starting HTTP Server
15/11/25 15:59:46 INFO server.Server: jetty-8.y.z-SNAPSHOT
15/11/25 15:59:46 INFO server.AbstractConnector: Started SocketConnector@0.0.0.0:56985
15/11/25 15:59:46 INFO util.Utils: Successfully started service 'HTTP file server' on port 56985.
15/11/25 15:59:46 INFO spark.SparkEnv: Registering OutputCommitCoordinator
15/11/25 15:59:46 INFO server.Server: jetty-8.y.z-SNAPSHOT
15/11/25 15:59:46 INFO server.AbstractConnector: Started SelectChannelConnector@0.0.0.0:4040
15/11/25 15:59:46 INFO util.Utils: Successfully started service 'SparkUI' on port 4040.
15/11/25 15:59:46 INFO ui.SparkUI: Started SparkUI at http://10.88.67.113:4040
15/11/25 15:59:46 INFO spark.SparkContext: Added JAR file:/home/hduser/HiBench/src/sparkbench/target/sparkbench-4.0-SNAPSHOT-MR2-spark1.5-jar-with-dependencies.jar at http://10.88.67.113:56985/jars/sparkbench-4.0-SNAPSHOT-MR2-spark1.5-jar-with-dependencies.jar with timestamp 1448447386404
15/11/25 15:59:46 WARN spark.SparkConf: The configuration key 'spark.kryoserializer.buffer.mb' has been deprecated as of Spark 1.4 and may be removed in the future. Please use spark.kryoserializer.buffer instead. The default value for spark.kryoserializer.buffer.mb was previously specified as '0.064'. Fractional values are no longer accepted. To specify the equivalent now, one may use '64k'.
15/11/25 15:59:46 WARN spark.SparkConf: The configuration key 'spark.kryoserializer.buffer.mb' has been deprecated as of Spark 1.4 and and may be removed in the future. Please use the new key 'spark.kryoserializer.buffer' instead.
15/11/25 15:59:46 WARN spark.SparkConf: The configuration key 'spark.kryoserializer.buffer.mb' has been deprecated as of Spark 1.4 and may be removed in the future. Please use spark.kryoserializer.buffer instead. The default value for spark.kryoserializer.buffer.mb was previously specified as '0.064'. Fractional values are no longer accepted. To specify the equivalent now, one may use '64k'.
15/11/25 15:59:46 WARN spark.SparkConf: The configuration key 'spark.kryoserializer.buffer.mb' has been deprecated as of Spark 1.4 and and may be removed in the future. Please use the new key 'spark.kryoserializer.buffer' instead.
15/11/25 15:59:46 WARN spark.SparkConf: The configuration key 'spark.kryoserializer.buffer.mb' has been deprecated as of Spark 1.4 and may be removed in the future. Please use spark.kryoserializer.buffer instead. The default value for spark.kryoserializer.buffer.mb was previously specified as '0.064'. Fractional values are no longer accepted. To specify the equivalent now, one may use '64k'.
15/11/25 15:59:46 WARN spark.SparkConf: The configuration key 'spark.kryoserializer.buffer.mb' has been deprecated as of Spark 1.4 and and may be removed in the future. Please use the new key 'spark.kryoserializer.buffer' instead.
15/11/25 15:59:46 WARN metrics.MetricsSystem: Using default name DAGScheduler for source because spark.app.id is not set.
15/11/25 15:59:46 INFO executor.Executor: Starting executor ID driver on host localhost
15/11/25 15:59:46 INFO util.Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 48605.
15/11/25 15:59:46 INFO netty.NettyBlockTransferService: Server created on 48605
15/11/25 15:59:46 INFO storage.BlockManagerMaster: Trying to register BlockManager
15/11/25 15:59:46 INFO storage.BlockManagerMasterEndpoint: Registering block manager localhost:48605 with 2.2 GB RAM, BlockManagerId(driver, localhost, 48605)
15/11/25 15:59:46 INFO storage.BlockManagerMaster: Registered BlockManager
15/11/25 16:01:15 WARN storage.MemoryStore: Not enough space to cache rdd_2_17 in memory! (computed 61.7 MB so far)
15/11/25 16:01:20 WARN storage.MemoryStore: Not enough space to cache rdd_2_18 in memory! (computed 61.7 MB so far)
15/11/25 16:01:24 WARN storage.MemoryStore: Not enough space to cache rdd_2_19 in memory! (computed 61.7 MB so far)
15/11/25 16:01:29 WARN storage.MemoryStore: Not enough space to cache rdd_2_20 in memory! (computed 61.7 MB so far)
15/11/25 16:01:33 WARN storage.MemoryStore: Not enough space to cache rdd_2_21 in memory! (computed 61.7 MB so far)
15/11/25 16:01:38 WARN storage.MemoryStore: Not enough space to cache rdd_2_22 in memory! (computed 61.7 MB so far)
15/11/25 16:01:42 WARN storage.MemoryStore: Not enough space to cache rdd_2_23 in memory! (computed 61.7 MB so far)
15/11/25 16:01:47 WARN storage.MemoryStore: Not enough space to cache rdd_2_24 in memory! (computed 61.7 MB so far)
15/11/25 16:01:51 WARN storage.MemoryStore: Not enough space to cache rdd_2_25 in memory! (computed 61.7 MB so far)
15/11/25 16:01:56 WARN storage.MemoryStore: Not enough space to cache rdd_2_26 in memory! (computed 61.7 MB so far)
15/11/25 16:02:00 WARN storage.MemoryStore: Not enough space to cache rdd_2_27 in memory! (computed 61.7 MB so far)
15/11/25 16:02:05 WARN storage.MemoryStore: Not enough space to cache rdd_2_28 in memory! (computed 61.7 MB so far)
15/11/25 16:02:09 WARN storage.MemoryStore: Not enough space to cache rdd_2_29 in memory! (computed 61.7 MB so far)
numExamples = 20000000.
15/11/25 16:02:33 WARN storage.MemoryStore: Not enough space to cache rdd_2_8 in memory! (computed 61.7 MB so far)
15/11/25 16:02:34 WARN storage.MemoryStore: Not enough space to cache rdd_2_8 in memory! (computed 11.5 MB so far)
15/11/25 16:03:17 WARN storage.MemoryStore: Not enough space to cache rdd_2_15 in memory! (computed 61.7 MB so far)
15/11/25 16:03:18 WARN storage.MemoryStore: Not enough space to cache rdd_2_15 in memory! (computed 11.5 MB so far)

**Comment by Jetsonpaul** [comment]
Running HiBench 5.0 with Spark 1.5.1 on 16GB RAM standalone machine.

**Comment by carsonwang** [comment]
Which data size were your running with? Were you able to run with a smaller data size? For example, set `hibench.scale.profile` to small.

**Comment by Jetsonpaul** [comment]
yes, for smaller data size we can able to run..!! Whether need to configure HiBench in 99-user_defined_properties.conf or Spark spark-defaults.conf files..??

**Comment by carsonwang** [comment]
Please set it in 99-user_defined_properties.conf. These settings will override the default settings.

**Comment by Jetsonpaul** [comment]
Thanks. Please close this issue.

**Comment by Jetsonpaul** [comment]
hibench.hadoop.home		    /usr/lib/hadoop
hibench.spark.home		    /home/hduser/spark-1.5.1
hibench.hdfs.master		     hdfs://localhost:9000
hibench.spark.master		    local[1]
hibench.hadoop.executable         /usr/lib/hadoop/bin/hadoop
hibench.hadoop.version		     hadoop2
hibench.spark.version                   spark1.5
spark.executor.memory                 8G
spark.driver.memory                     12G
spark.storage.memoryFraction      1
spark.kryoserializer.buffer.mb	       2000

i set in 99-user_defined_properties.conf file, even though same issue...

**Comment by carsonwang** [comment]
You said the smaller data size works. What do you mean by saying "same issue"?

**Comment by Jetsonpaul** [comment]
Without checked log details i said wrongly, log file which contain `WARN storage.MemoryStore: Not enough space to cache rdd`

**Comment by carsonwang** [comment]
I think your configuration `spark.storage.memoryFraction` was causing the problem. Can you please remove that. By default the value is `0.6`. If you set it to 1, Spark will have no memory for other executions.

**Comment by Jetsonpaul** [comment]
But first i kept with default `spark.storage.memoryFraction` `0.6` but got this error, then read `spark configuration documentation` then only i set into `1`

**Comment by carsonwang** [comment]
No, you can't set it to 1. Are you sure you are running with the `small` data size? Can you give more memory to the executor and less to the driver. For example give 16G to the executor and 4G to the driver.

**Comment by Jetsonpaul** [comment]
ok but i gave spark.executor.memory `8G` spark.driver.memory `12G`..  is it right way to set memory or not..??

**Comment by carsonwang** [comment]
Yes you can set them in  99-user_defined_properties.conf. The driver doesn't require so many memory. In most of the cases, 4G is enough for the driver. You can give more to the executor.

**Comment by Jetsonpaul** [comment]
thanks for quick reply, let me try and let u know soon..
